Remove database consistency validation of clinical attributes

fedde-s opened the pull request:
Clinical attributes used to share a single namespace across all studies, meaning that once the meaning of an attribute name was defined (with a display name, mouseover description, data type, sample or patient level and priority), all subsequent studies would have to adhere to that. Commit b6dcfc082116293178b0cd762b0be7aba3152d51 made clinical attributes study-specific. Because of that, validating whether the definitions of attributes in a study are consistent with those configured in the portal is now no longer applicable. Removing the dependency on these APIs fixes the problems caused by duplicate entries introduced into this API's output, reported in issue #1804.

Changes proposed in this pull request:
- Remove database consistency validation for clinical attributes
- Remove clinical attribute API calls from the validator
- Clarify the message emitted when attributes do break requirements by adding the line and column numbers of the offending value

fedde-s commented:
@jjgao: This is the fix Pieter asked about in the **etl** channel on Slack. It might be easiest to merge pull request #1809 to `hotfix` first, as these are the changes that have already been approved as part of pull request #1567. After that, the only additional changes to be reviewed in this pull request will be the three commits that fix issue #1804.

fedde-s commented:
I will replace the first five commits (the ones in #1809) by the original commits that are already in rc, to prevent merging issues.

fedde-s commented:
Done.

pieterlukasse commented on core/src/main/scripts/importer/validateData.py:
```diff
@@ -1284,24 +1283,146 @@ class ClinicalValidator(Validator):
+
+    # attributes required to have certain properties because of hard-coded use
+    # TODO add unit tests for this functionality
+    PREDEFINED_ATTRIBUTES = {
```
this is repeating what is already in #1809

fedde-s replied:
You are looking at the exact same commit.

pieterlukasse commented on core/src/main/scripts/importer/validateData.py:
```diff
@@ -1569,7 +1564,7 @@ def checkHeader(self, cols):
                                        'column_number': col_index + 1,
                                        'cause': col_name})
-                    else:
+                    elif not self.relaxed_mode:
```
please add some comments here for why this is not applicable to relaxed_mode

pieterlukasse commented on core/src/main/scripts/importer/validateData.py:
```diff
@@ -1640,23 +1593,15 @@ def checkLine(self, data):
                 if not self.checkFloat(value):
                     self.logger.error(
-                        'Value of attribute to be loaded as NUMBER is not a real number' + according_to_portal,
+                        'Value of attribute to be loaded as NUMBER is not a real number',
```
change message to  'Value of attribute is not a real number'

pieterlukasse commented on core/src/main/scripts/importer/validateData.py:
```diff
@@ -1770,11 +1715,11 @@ def checkHeader(self, cols):
                        'cause': 'SAMPLE_ID'})
         # refuse to define attributes also defined in the sample-level file
-        for new_attribute in self.newly_defined_attributes:
+        for new_attribute in self.defined_attributes:
```
`for defined_attribute in self.defined_attributes`?

pieterlukasse commented on core/src/test/scripts/unit_tests_validate_data.py:
```diff
@@ -1298,26 +1283,27 @@ def test_unnecessary_and_wrong_stable_id(self):
+        # 1 for CANCER_TYPE being in DEFINED_SAMPLE_ATTRIBUTES
+        # 2 for missing survival status attributes and
+        # 4 for patients with no samples
+        self.assertEqual(len(record_list), 9)
```
this is better implemented using the error file diff at the end

fedde-s replied:
I agree. But the design of this test case is not the issue I intended to fix in this pull request.

pieterlukasse commented:
Look very good :+1:  Just some minor comments.

fedde-s commented:
Thank you for the review, @pieterlukasse. I addressed most of your comments, Although the last two, I'd argue are a separate (though minor) issue introduced in pull request #1581.

fedde-s commented:
...and now I fixed the test that started failing because one of the amendments changed an output message.

fedde-s commented:
As @pieterlukasse requested, I added three commits to clarify the purpose of a few automated tests I edited and adjust their assertions to better match this purpose.

pieterlukasse commented on core/src/test/scripts/system_tests_validate_data.py:
```diff
@@ -47,24 +39,24 @@ def tearDown(self):
+                # generate a diff between the expected and the generated file,
+                # even if the diff is longer than a few lines
+                self.maxDiff = None
+                self.assertMultiLineEqual(ref_file.read(), out_file.read())
```
works :+1:
